app/models/oai_set.rb

```diff
@@ -37,12 +37,21 @@ def sets_from_facets(facets)
   def initialize(spec, opts = {})
     @spec = spec
     @name = opts[:name] || name_from_spec
-    @description = opts[:description]
+    @description = opts[:description] || description_from_spec
   end
 
   private
 
   def name_from_spec
     ActiveFedora::SolrService.query("has_model_ssim:AdminSet AND id:#{@spec}", rows: 1).first['title_tesim'].first
   end
+
+  # METHOD: Add in an option to fetch the metadata description to add onto OAI
+  def description_from_spec
+    # QUERY: Use the query to get the description per collection using save navigation
+    describe_meta = ActiveFedora::SolrService.query("has_model_ssim:AdminSet AND id:#{@spec}", rows: 1).first['description_tesim']&.first
+
+    # CONDITION: Check the condition to return the correct description to display on OAI
+    return describe_meta if describe_meta != '' || describe_meta.nil?
```

Comment:
Check the functionality of `#blank?` or `#empty?`. I'm not sure which works better in this case

Reply:
The functionality `blank?` would work and work on `nil` as well but not for `empty?` but I'm currently writing in a safe guard statement due to RuboCop hate if I write if-else block. But when i tried `blank?` only the one with the `[""]` will print out and nothing else it is print out.

Reply:
Got it fixed and pushing it as a PR.